### PR TITLE
Interlinks: Improve HTML tag handling. Fixes #885

### LINKED ISSUE #885: [Bug] [Interlinks] Incorect details and summary tags conversion
### 1. Summary

If I use [**Interlinks**](https://github.com/getpelican/pelican-plugins/tree/master/interlinks) plugin, [**`<details>` and `<summary>` tags**](http://www.hongkiat.com/blog/html5-details-summary-tags/) incorrect convert for me.

### 2. Settings

Part of my `pelicanconf.py` file:

```python
PLUGIN_PATHS = ['pelican-plugins']
PLUGINS = [
    'interlinks',
]

INTERLINKS = {
    'kristinita': 'http://kristinita.ru/#gsc.tab=0&gsc.q='
}
```

Part of my `Kristinita.md` file:

```html
<details><summary>Santa</summary>

Kristina</details>
```

### 3. Steps to reproduce

I build my site use any command, for example:

```shell
pelican content
```

### 4. Expected behavior

If Interlinks disable:

```html
<p><details><summary>Santa</summary></p>
<p>Kristina</details></p>
```

### 5. Actual behavior

If Interlinks enable:

```html
<p><details><summary>Santa</summary></details></p>
<p>Kristina</p>
```

### 6. Environment

**Operating system and version:**
Windows 10 Enterprise LTSB 64-bit EN
**Python:**
3.6.1
**Pelican:**
3.7.1
**Python Markdown:**
2.6.8
**BeautifulSoup4**
4.5.3

Thanks.

### PR DESCRIPTION
**Pull request by aquinzi**
Fixes bad parsing of tags when trying to convert a/img tags.

### CODE REVIEW

**Comment by justinmayer** [comment]
Many thanks to @kristinita for the detailed bug report and to @aquinzi for the fix!  (^_^)

**Comment by andyli** [comment]
This broke https://github.com/getpelican/pelican-plugins/pull/406

File: interlinks/interlinks.py [hunk 1]
```diff
@@ -23,32 +24,39 @@ def getSettings (generator):
...
+			text = BeautifulSoup(content, "html.parser", parse_only=SoupStrainer("img"))
 			for img in text.find_all('img', src=re.compile("(.+?)>")):
+				old_tag = str(img)
 				url = img.get('src')
 				m = re.search(r"(.+?)>", url).groups()
 				name = m[0]
 				if name in interlinks:
 					hi = url.replace(name+">",interlinks[name])
 					img['src'] = hi
+				content = content.replace(old_tag, str(link))
```
Comment (andyli) [inline interlinks/interlinks.py]:
```
UnboundLocalError: local variable 'link' referenced before assignment
```

`link` should be `img`.

Comment (andyli) [inline interlinks/interlinks.py]: Another problem. `content = content.replace(old_tag, ...)` really wouldn't work.

To illustrate, given the markdown
```markdown
![Andy Li](this>files/profile-s.png)
```
The output will be
```html
<img alt="Andy Li" src="this&gt;files/profile-s.png">
```

But `old_tag = str(img)` will output
```html
<img alt="Andy Li" src="this&gt;files/profile-s.png"/>
```

Notice `` vs ``...
So, the replacement wouldn't happen...

**Comment by aquinzi** [comment]
Uploaded a fix

**Pull request merged**